api/src/types/IRead.ts

```diff
@@ -1,4 +1,9 @@
+export interface IFindOneOptions {
+    id?: string;
+    slug?: string;
+}
+
 export default interface IRead<T> {
     findAll(): Promise<T[]>;
-    findOne(id: string): Promise<T | null>;
+    findOne(options: IFindOneOptions): Promise<T | null>;
```

Comment:
This does not need to be an object. Change it back to taking only : id: string. If we need to worry about getting by various possible fields, we can do that later. For now, we want only a single identifier to be used. For posts, it's slug, for most others, it will just be their id.